tests/MoBi.Tests/Core/SBML/UnitImporterSpecs.cs

```diff
@@ -17,95 +17,78 @@ namespace MoBi.Core.SBML
-
-            _unitDef3 = _sbmlModel.createUnitDefinition();
-            _unitDef3.setId("inverse_day");
-            _unitDef3.setName("1/d");
-            _unitDef3.addUnit(_unit3);
-
-
-            _sbmlModel.addUnitDefinition(_unitDef);
-            sut.DoImport(_sbmlModel, new MoBiProject(), A.Fake<SBMLInformation>(), new MoBiMacroCommand());
-            _volumeDimension = _dimensionFactoy.Dimension("Volume");
-         }
-
-         [Observation]
-         public void NotNullTest()
-         {
-            _volumeDimension.ShouldNotBeNull();
-         }
-
-         [Observation]
-         public void UnitImporterTest()
-         {
-            _volumeDimension.ShouldNotBeNull();
-            _volumeDimension.Name.ShouldBeEqualTo("Volume");
-            _volumeDimension.DefaultUnit.Name.ShouldBeEqualTo("l");
-         }
-
-         [Observation]
-         public void NewDimensionsTests()
-         {
-            sut.ToMobiBaseUnit("substance", 3e6).value.ShouldBeEqualTo(3e12);
-            sut.ToMobiBaseUnit("inverse_day", 0.83).value.ShouldBeEqualTo(0.83/1440); //base unit is in minutes so 1440 minutes in one day
-         }
+         sut = IoC.Resolve<IUnitDefinitionImporter>();
+         //create unit "substance"
+         _unit = new Unit(3, 1);
+         _unit.setExponent(1);
+         _unit.setMultiplier(1.0);
+         _unit.setScale(0);
+         _unit.setKind(libsbml.UNIT_KIND_MOLE);
+
+         //create unitdef "substance"
+         _unitDef = _sbmlModel.createUnitDefinition();
+         _unitDef.setId("substance");
+         _unitDef.setName("substance");
+         _unitDef.addUnit(_unit);
+
+         //create unit "volume"
+         _unit2 = _sbmlModel.createUnit();
+         _unit2.setExponent(1);
+         _unit2.setMultiplier(1.0);
+         _unit2.setScale(0);
+         _unit2.setKind(libsbml.UnitKind_forName("volume"));
+
+         //create unitdef "volume"
+         _unitDef2 = _sbmlModel.createUnitDefinition();
+         _unitDef2.setId("volume");
+         _unitDef2.setName("volume");
+         _unitDef2.addUnit(_unit2);
+
+         _unit3 = _sbmlModel.createUnit();
+         _unit3.setExponent(-1);
+         _unit3.setMultiplier(86400.0); //86400 seconds in one day
+         _unit3.setScale(0);
+         _unit3.setKind(libsbml.UnitKind_forName("second"));
+
+         _unitDef3 = _sbmlModel.createUnitDefinition();
+         _unitDef3.setId("inverse_day");
+         _unitDef3.setName("1/d");
+         _unitDef3.addUnit(_unit3);
+
+
+         _sbmlModel.addUnitDefinition(_unitDef);
+         sut.DoImport(_sbmlModel, new MoBiProject(), A.Fake<SBMLInformation>(), new MoBiMacroCommand());
+      }
+
+      [Observation]
+      public void NewDimensionsTests()
+      {
+         sut.ToMobiBaseUnit("substance", 3e6).value.ShouldBeEqualTo(3e12);
+         sut.ToMobiBaseUnit("inverse_day", 0.83).value.ShouldBeEqualTo(0.83 / 1440); //base unit is in minutes so 1440 minutes in one day
```

Comment:
but isn't the value in second based on _unit3?

Reply:
Minutes is the base unit, right?